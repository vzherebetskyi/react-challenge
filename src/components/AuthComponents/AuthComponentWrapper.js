import React from 'react';

import { MainLogoImg } from '../../assets/images/images';
import QuestionImg from '../../assets/images/question.svg';

const AuthComponentWrapper = ({ children, title, history }) => (
  <div className="auth-container">
    <div className="auth-title-panel">
      <img src={MainLogoImg} alt="main-logo" />
    </div>
    <div onClick={() => history.push('/faq')} className="auth-faq-button">
      <img src={QuestionImg} alt="question" />
    </div>
    <div className="form-container">
      <h1 className="base-title">{title}</h1>
      {children}
    </div>
  </div>
);

export default AuthComponentWrapper;
