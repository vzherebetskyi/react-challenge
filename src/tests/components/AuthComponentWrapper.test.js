import React from 'react';
import renderer from 'react-test-renderer';

import AuthComponentWrapper from '../../components/AuthComponents/AuthComponentWrapper';

const history = { push: jest.fn() };

test('should render AuthComponentWrapper correctly', () => {
  const elem = renderer
    .create(
      <AuthComponentWrapper title="LOGIN" history={history}>
        <div />
      </AuthComponentWrapper>
    )
    .toJSON();
  expect(elem).toMatchSnapshot();
});
