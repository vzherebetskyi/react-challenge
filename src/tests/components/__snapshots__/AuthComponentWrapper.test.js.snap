// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render AuthComponentWrapper correctly 1`] = `
<div
  className="auth-container"
>
  <div
    className="auth-title-panel"
  >
    <img
      alt="main-logo"
      src={Object {}}
    />
  </div>
  <div
    className="auth-faq-button"
    onClick={[Function]}
  >
    <img
      alt="question"
      src={Object {}}
    />
  </div>
  <div
    className="form-container"
  >
    <h1
      className="base-title"
    >
      LOGIN
    </h1>
    <div />
  </div>
</div>
`;
